src/MassTransit.Abstractions/Attributes/MessageUrnAttribute.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace MassTransit.Attributes
+{
+    using System;
+
+    /// <summary>
+    /// Specify the URN used for this message contract
+    /// if configured.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public class MessageUrnAttribute :
+        Attribute
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="urn">The urn to use for this message type.</param>
+        public MessageUrnAttribute(string urn)
+        {
+            Urn = urn;
```

Comment:
would like to see some validation that when formatted as a Urn it doesn't blow up.

Reply:
I've updated the attribute to perform validation when it is initialized.  This is to ensure the error throws as early as possible, identifying the error as an an issue in the attribute itself, not just a run time error in resolving the urn.